modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/sqlquery/DbExecutor.scala

```diff
@@ -82,8 +83,8 @@ object DbExecutor {
 
   implicit def syncDbExecutor[F[_]: Sync]: DbExecutor[F] =
     new DbExecutor[F] {
-      def getConnection(dataSource: DataSource): F[Either[Throwable, Connection]] =
-        Sync[F].delay(Either.catchNonFatal(dataSource.getConnection()))
+      def getConnection(dataSource: DataSource, blocker: BlockerF[F]): F[Either[Throwable, Connection]] =
+        blocker.blockOn(Sync[F].delay(Either.catchNonFatal(dataSource.getConnection())))
```

Comment:
`Sync` (for synchronous) means that the thread executing the task will wait for its result before moving on. While it waits for its result, it does nothing else. So in this case, while the connection is acquired, the thread will waste processing time doing nothing.

`Blocker` can be seen as an other thread pool dedicated to blocking tasks (I/O bound tasks). Everything that involves some I/O should be executed inside `Blocker.blockOn`, so that the main thread-pool (used for CPU-bound tasks) stays fully utilized and never wastes time waiting.

Reply:
Ah that makes total sense, thanks for explaining!